pkg/server/clock_monotonicity.go

```diff
@@ -187,16 +186,19 @@ func periodicallyPersistHLCUpperBound(
 
 	for {
 		select {
-		case persistInterval, ok = <-persistHLCUpperBoundIntervalCh:
-			ticker.Stop()
-			if !ok {
-				return
+		case updatedPersistInterval := <-persistHLCUpperBoundIntervalCh:
+			if updatedPersistInterval == persistInterval {
+				// No change.
+				continue
 			}
+			persistInterval = updatedPersistInterval
 
+			ticker.Stop()
 			if persistInterval > 0 {
 				ticker = tickerFn(persistInterval)
 				persistHLCUpperBound()
-				log.Ops.Info(context.Background(), "persisting HLC upper bound is enabled")
+				log.Ops.Infof(context.Background(), "persisting HLC upper bound is enabled [every %.2fs]",
```

Comment:
Not for this PR, but I think we could flow workerCtx through to this function to have log tags.